app/models/sensor.rb

```diff
@@ -1,6 +1,7 @@
 class Sensor < ApplicationRecord
   has_many :measurements, dependent: :destroy
   belongs_to :sponsor, optional: true
+  belongs_to :waterbody, optional: true
```

Comment:
If it's non-optional, how are migrations done?

I think it might still be useful to have sensors without a waterbody. This could be used e.g. for test sensors. But I don't have a strong opinion on that, if it makes things easier to require a waterbody, that's fine as well.

Reply:
> If it's non-optional, how are migrations done?

By providing a default waterbody. (something like "unknown" or similar)

> I think it might still be useful to have sensors without a waterbody. This could be used e.g. for test sensors. But I don't have a strong opinion on that, if it makes things easier to require a waterbody, that's fine as well.

It's easier to just have it optional.